### PR TITLE
Add WorkflowChain class and valid data example

### PR DESCRIPTION
**Pull request by mbthornton-lbl**
- update to workflow_execution_activity.yaml to add class
- new valid/WorkflowChain.yaml
- new invalid data examples

### CODE REVIEW

File: src/schema/workflow_execution_activity.yaml [hunk 1]
```diff
@@ -23,6 +23,30 @@ subsets:
       Subset consisting of just the workflow execution activities
 
 classes:
+  WorkflowChain:
+    description: >-
+      A workflow chain is a sequence of workflow execution activities that are related to each other in some way.
```
Comment (brynnz22) [inline src/schema/workflow_execution_activity.yaml:28]: change sequence to collection

Comment (brynnz22) [reply src/schema/workflow_execution_activity.yaml:28]: Add the sequence of collection can be discover by has_input and has_output patterns

**Comment by mslarae13** [comment]
InstrumentData is a DataObject and it is the output from DataGeneration and that InstrumentData is the input for WorkflowChain & the first WorkflowExecution processes 

1- link WorkflowChain directly to the data generation process
2- link to the first steps in the chain
3- link to the outputs of the data generation process

File: src/schema/workflow_execution_activity.yaml [hunk 2]
```diff
@@ -23,6 +23,30 @@ subsets:
...
+    slots:
+      - id
+      - type
+      - analyte_category
+      - part_of
```
Comment (brynnz22) [inline src/schema/workflow_execution_activity.yaml:33]: remove part_of make has_input that is the data object?

File: src/data/valid/WorkflowChain.yaml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+id: nmdc:wfch-11-ab
+type: nmdc:WorkflowChain
+analyte_category: metagenome
```
Comment (brynnz22) [inline src/data/valid/WorkflowChain.yaml:3]: computational protocol?

**Comment by anastasiyaprymolenna** [comment]
if WorkflowChain considered a process? It should inherit from PlannedProcess, instead of NamedThing if that is the case

**Comment by brynnz22** [comment]
Ah good catch! It should be changed now.

File: src/schema/workflow_execution_activity.yaml [hunk 4]
```diff
@@ -23,6 +23,39 @@ subsets:
...
 classes:
+  WorkflowChain:
+    is_a: PlannedProcess
+    description: >-
+      A workflow chain is a collection of Workflow Execution(s) that 
+      analyse the output of DataGeneration.
+    comments: >-
+      The sequence of Workflow Execution(s) can be inferred from the has_input and has_output slots of the Workflow Execution(s)
+    slots:
+      - id
```
Comment (mbthornton-lbl) [inline src/schema/workflow_execution_activity.yaml:34]: delete me

File: src/schema/workflow_execution_activity.yaml [hunk 5]
```diff
@@ -23,6 +23,39 @@ subsets:
...
+      id:
+        required: true
+        structured_pattern:
+          syntax: "{id_nmdc_prefix}:wfch-{id_shoulder}-{id_blade}{id_version}{id_locus}"
+          interpolated: true
+      type:
+        required: true
+      analysis_category:
+        required: true
+      analyzes_data_from:
```
Comment (mbthornton-lbl) [inline src/schema/workflow_execution_activity.yaml:49]: was_informed_by

File: src/schema/workflow_execution_activity.yaml [hunk 6]
```diff
@@ -314,8 +353,32 @@ classes:
           syntax: "{id_nmdc_prefix}:wfnom-{id_shoulder}-{id_blade}{id_version}{id_locus}"
           interpolated: true
 
+enums:
+  analysis_category_enum:
+    from_schema: https://example.com/nmdc_dh
```
Comment (mbthornton-lbl) [inline src/schema/workflow_execution_activity.yaml:358]: re-use AnalyteCategoryEnum in nmdc.yaml

File: src/schema/workflow_execution_activity.yaml [hunk 7]
```diff
@@ -314,8 +353,32 @@ classes:
...
 slots:
+  analysis_category:
+    description: The category of analysis performed
+    examples:
+      - value: metagenomics
+    see_also:
+      - MIxS:investigation_type
+    range: analysis_category_enum
+
+  analyzes_data_from:
```
Comment (mbthornton-lbl) [inline src/schema/workflow_execution_activity.yaml:375]: was_informed_by

File: src/schema/workflow_execution_activity.yaml [hunk 8]
```diff
@@ -314,8 +353,32 @@ classes:
...
+    from_schema: https://example.com/nmdc_dh
+    permissible_values:
+      metabolomics: { }
+      metagenomics: { }
+      metaproteomics: { }
+      metatranscriptomics: { }
+      natural organic matter: { }
 
 slots:
+  analysis_category:
```
Comment (mbthornton-lbl) [inline src/schema/workflow_execution_activity.yaml:367]: use analyte_category - remove redundant slot defs

**Review by turbomam — APPROVED** [review]

**Pull request merged**